cmd/binapi-generator/main.go

```diff
@@ -31,40 +29,53 @@ import (
 
 func init() {
 	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, "USAGE\n")
-		fmt.Fprintf(os.Stderr, "  Parse API_FILES and generate Go bindings\n")
-		fmt.Fprintf(os.Stderr, "  Provide API_FILES by file name, or with full path including extension.\n")
-		fmt.Fprintf(os.Stderr, "  %s [OPTION] API_FILES\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "Generating Go bindings for the VPP API\n")
+		fmt.Fprintf(os.Stderr, "--------------------------------------\n\n")
+		fmt.Fprintf(os.Stderr, "This generates VPP api Go bindings based on .api.json files or a VPP repository\n\n")
+		fmt.Fprintf(os.Stderr, "Usage: %s [OPTION]\n", os.Args[0])
 		fmt.Fprintf(os.Stderr, "OPTIONS\n")
 		flag.PrintDefaults()
 		fmt.Fprintf(os.Stderr, "\nEXAMPLES:\n")
-		fmt.Fprintf(os.Stderr, "  %s \\\n", os.Args[0])
-		fmt.Fprintf(os.Stderr, "    --input-dir=$VPP/build-root/install-vpp-native/vpp/share/vpp/api/ \\\n")
-		fmt.Fprintf(os.Stderr, "    --output-dir=~/output \\\n")
-		fmt.Fprintf(os.Stderr, "    interface ip\n")
-		fmt.Fprintf(os.Stderr, "  Assuming --input-dir contains interface.api.json & ip.api.json\n")
+		fmt.Fprintf(os.Stderr, "  Generate bindings from VPP API files (*.api.json):\n")
+		fmt.Fprintf(os.Stderr, "   %s --api /usr/share/vpp/api --output ./myapp/binding --filter interface,ip\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "  Generate bindings from the VPP repository :\n")
+		fmt.Fprintf(os.Stderr, "   %s --vpp ~/vpp --output ./myapp/binding\n", os.Args[0])
 	}
 }
 
 func printErrorAndExit(msg string) {
-	fmt.Fprintf(os.Stderr, "Error: %s\n\n", msg)
+	color.New(color.FgRed).Fprintf(os.Stderr, "Error:\n%s\n\n", msg)
 	flag.Usage()
 	os.Exit(1)
 }
 
 func main() {
-	var (
-		theApiDir        = flag.String("input-dir", vppapi.DefaultDir, "Input directory containing API files. (e.g. )")
-		theOutputDir     = flag.String("output-dir", "binapi", "Output directory where code will be generated.")
-		importPrefix     = flag.String("import-prefix", "", "Prefix imports in the generated go code. \nE.g. other API Files (e.g. api_file.ba.go) will be imported with :\nimport (\n  api_file \"<import-prefix>/api_file\"\n)")
-		generatorPlugins = flag.String("gen", "rpc", "List of generator plugins to run for files.")
-		theInputFile     = flag.String("input-file", "", "DEPRECATED: Use program arguments to define files to generate.")
-
-		printVersion     = flag.Bool("version", false, "Prints version and exits.")
-		debugLog         = flag.Bool("debug", false, "Enable verbose logging.")
-		noVersionInfo    = flag.Bool("no-version-info", false, "Disable version info in generated files.")
-		noSourcePathInfo = flag.Bool("no-source-path-info", false, "Disable source path info in generated files.")
-	)
+	apiSrcDir := flag.String("input-dir", vppapi.DefaultDir, "[DEPRECATED, use --api] Input directory containing API files.")
+	flag.StringVar(apiSrcDir, "api", "", "Generate based on .api files in this directory.")
```

Comment:
The API directory and VPP directory could be resolved automatically from the `input-dir` without needing user to set them in different parameters.

Let's say we rename the `input-dir` to `input` to make it shorter and resolve directories like this:
- if `input` is empty (default)
  - check if VPP is installed
  - load API files from installed VPP or return error if not installed
  - generate
- if `input` points to a directory with `core` and `plugins` directories 
  - then assume it is the **API directory**  
  - load API files from `input`
  - generate
- if `input` points to local VPP directory (cloned git repository)
  - then assume it is the **VPP directory** we do development on
  - run `make json-api-files`
  - load API files from `build-root/install-vpp-native/vpp/share/vpp/api/` (relative to `input`)
  - generate